worker/src/queue.rs

```diff
@@ -1,124 +1,109 @@
-use std::marker::PhantomData;
-
 use crate::{env::EnvBinding, Date, Error, Result};
 use js_sys::Array;
 use serde::{de::DeserializeOwned, Serialize};
 use wasm_bindgen::{prelude::*, JsCast};
 use wasm_bindgen_futures::JsFuture;
-use worker_sys::{MessageBatch as MessageBatchSys, Queue as EdgeQueue};
-
-static BODY_KEY_STR: &str = "body";
-static ID_KEY_STR: &str = "id";
-static TIMESTAMP_KEY_STR: &str = "timestamp";
+use worker_sys::{Message as MessageSys, MessageBatch as MessageBatchSys, Queue as EdgeQueue};
 
-pub struct MessageBatch<T> {
+/// A batch of messages that are sent to a consumer Worker.
+pub struct MessageBatch {
```

Comment:
I'm not too fond of the removal of the body type generic. I get that we want to support multiple types for the body but that's currently possible using [serde's enum representations](https://serde.rs/enum-representations.html#untagged) unless theres some edge case I'm not aware of. Besides that I think it's good that we stay in-line with the TypeScript type definition that does use a generic.

Reply:
Thanks for the reviewing.

I've changed it back to `MessageBatch<T>` and also re-added the `messages()` fn, so it'll match the typescript types.

The main change is now the `MessageBatch<T>` does require `T: DeserializeOwned`, this means that the `iter()` fn now gives `Item=RawMessage`. If you don't like this, happy to I can re-add the `DeserializeOwned` bound and remove the whole `RawMessage` part.

Reply:
I think the best approach to keep both the flexibility and the ease of use is to rename the current `iter` method to `raw_iter()` (and rename `MessageIter` to `RawMessageIter`) and then add:

```rs
impl<T: DeserializeOwned> MessageBatch<T> {
    /// Iterator for messages in the message batch. Ordering of messages is not guaranteed.
    pub fn iter(&self) -> MessageIter<T> {
        let messages = self.inner.messages();
        MessageIter {
            range: 0..messages.length(),
            array: messages,
        }
    }
}
```

Reply:
Good idea, I've updated it to follow your suggestion.